packages/wallet/contract/src/smart-wallet.js

```diff
@@ -0,0 +1,80 @@
+// @ts-check
+import { E, Far } from '@endo/far';
+import {
+  makeStoredSubscription,
+  makeSubscriptionKit,
+  observeIteration,
+} from '@agoric/notifier';
+import '@agoric/deploy-script-support/exported.js';
+import '@agoric/zoe/exported.js';
+import spawn from '@agoric/wallet-backend/src/wallet.js';
+
+import '@agoric/wallet-backend/src/types.js'; // TODO avoid ambient types
+
+const { assign, entries, keys, fromEntries } = Object;
+
+/**
+ *
+ * TODO: multi-tennant wallet
+ *
+ * @param {ZCF} zcf
+ * @param {{
+ *   storageNode?: ERef<StorageNode>,
+ *   marshaller?: ERef<Marshaller>,
+ * }} privateArgs
+ */
+export const start = async (zcf, privateArgs) => {
```

Comment:
we need some test of this contract starting and the shape of its `creatorFacet`. I can help with that if you like.